docs/dev/agent_api.md

```diff
@@ -1,17 +1,89 @@
-# IPC (Inter Process Communication) API
+# Endpoints exposed by the Agent
+The core Agent exposes a variety of HTTP and GRPC endpoints that can be grouped into 2
+groups:
+1) Control
+    - These endpoints are used to send commands that can control and inspect the state of the running Agent
+2) Telemetry
+    - These expose some internal telemetry that is useful for profiling and
+      debugging.
 
-The agent communicates with the outside world through an HTTP API to ease the
-development of 3rd party tools and interfaces. The API is available from `localhost`
-and through HTTPS only. It listens on port `5001` by default but can be configured differently.
+## Control API
+This API is accessible via HTTPS only and listens by default on the `localhost` interface on port `5001`. The listening interface and port can be configured using the `cmd_host` and `cmd_port` config options.
 
-## Security and Authentication
+### Authentication
+To avoid unprivileged users accessing the Agent control API, authentication is required and based on a generated token.
+The token is written to a file (`auth_token`) that's only readable by the user that the Agent runs as.
 
-To avoid unprivileged users to access the API, authentication is required and based on a token.
-The token is written to a file that's only readable by the user that the Agent runs as.
+The `auth_token` file is written to the same directory that the config was read
+from by default, or the config option `auth_token_file_path` can be used to set
+a custom location.
 
-## Endpoints
+### Example
+```
+$ curl -qs -H "authorization: Bearer $(cat /path/to/auth_token)" -k https://localhost:5001/agent/version
+{"Major":7,"Minor":41,"Patch":0,"Pre":"rc.3","Meta":"git.238.453e769","Commit":"453e7695a4"}%
+```
+
+### Full Endpoint List
+https://github.com/DataDog/datadog-agent/blob/453e7695a43fa3c162e1240863999c9ddc91fbdd/cmd/agent/api/internal/agent/agent.go#L49-L73
```

Comment:
The inline code preview is cool! I didn't know github did this.

Reply:
Same! Was pleasantly surprised here